Dockerfile-ubuntu.template

```diff
@@ -194,37 +178,48 @@ RUN set -eux; \
 		--without-wx \
 		$jitFlag \
 	; \
+	\
 # Compile & install Erlang/OTP
 	make -j "$(getconf _NPROCESSORS_ONLN)" GEN_OPT_FLGS="-O2 -fno-strict-aliasing"; \
 	make install; \
-	cd ..; \
-	rm -rf \
-		"$OTP_PATH"* \
-		/usr/local/lib/erlang/lib/*/examples \
-		/usr/local/lib/erlang/lib/*/src \
-	; \
 	\
-# reset apt-mark's "manual" list so that "purge --auto-remove" will remove all build dependencies
-	apt-mark auto '.*' > /dev/null; \
-	[ -z "$savedAptMark" ] || apt-mark manual $savedAptMark; \
-	find /usr/local -type f -executable -exec ldd '{}' ';' \
+# Remove unnecessary files
+	find /usr/local/lib/erlang -type d -name examples -exec rm -rf '{}' +; \
+	find /usr/local/lib/erlang -type d -name src -exec rm -rf '{}' +; \
+	find /usr/local/lib/erlang -type d -name include -exec rm -rf '{}' +
+
+# Check that Erlang/OTP crypto & ssl were compiled against OpenSSL correctly
+RUN erl -noshell -eval 'io:format("~p~n~n~p~n~n", [crypto:supports(), ssl:versions()]), init:stop().'
+
+FROM ubuntu:20.04
+
+COPY --from=erlang-builder /usr/local/bin/ /usr/local/bin/
+COPY --from=erlang-builder /usr/local/etc/ssl/ /usr/local/etc/ssl/
+COPY --from=erlang-builder /usr/local/lib/ /usr/local/lib/
+
+ENV RABBITMQ_DATA_DIR=/var/lib/rabbitmq
+
+RUN set -eux; \
+# Configure OpenSSL to use system certs
+	ln -vsf /etc/ssl/certs /etc/ssl/private /usr/local/etc/ssl; \
+	\
+# Ensure run-time dependencies are installed
+	find /usr/local -type f -executable -exec ldd '{}' ';' 2>/dev/null \
 		| awk '/=>/ { print $(NF-1) }' \
 		| sort -u \
-		| xargs -r dpkg-query --search \
+		| xargs -r dpkg-query --search 2>/dev/null \
 		| cut -d: -f1 \
 		| sort -u \
-		| xargs -r apt-mark manual \
+		| xargs -r apt-get install --yes --no-install-recommends \
 	; \
```

Comment:
Sorry, I should've been more specific -- if we ever stop getting lucky and not having our deps in the base image already, this won't quite work as-is because `ldd` won't list the ones we're missing in the same way it did while the deps are installed, _and_ we can't then `dpkg-query --search` to ask which package they come from without having them installed, so we'd need to do something like gather the list of APT dependencies in the previous stage into a text file or something so we can install it here.

(On the other hand, Alpine doesn't suffer from this because it uses `readelf` to get a list of `.so` dependencies and the package system there includes the `so:...` aliases we take advantage of, so it "just works" appropriately there in ways Ubuntu can't. :sweat_smile:)

I'm totally on board with just dropping this completely from Ubuntu for now (since our `openssl version` and/or `erl` tests below will catch any problems and fail before we publish broken images), and then we revisit the implementation later, or if you wanted to pick an appropriate place to have the previous stage write an `apt-deps.txt` or similar that we can do something like `xargs < /.../apt-deps.txt -rt apt-get install --yes --no-install-recommends` on here.  Your choice! :smile:

Reply:
Well that's interesting. Seems like the simplest route is to just nuke it and let the tests catch it.